hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java

```diff
@@ -55,52 +61,115 @@
+        return mapConfigs.stream().map(MapConfig::getWanReplicationRef)
+                .filter(Objects::nonNull).count();
     }
 
     private long countMapWithAtleastOneAttribute() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getAttributeConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getAttributeConfigs)
+                .filter(attributeConfigs -> !(attributeConfigs.isEmpty())).count();
     }
 
     private long countMapUsingEviction() {
-        return mapConfigs.stream().filter(mapConfig -> mapConfig != null
-                && mapConfig.getEvictionConfig().getEvictionPolicy() != EvictionPolicy.NONE).count();
+        return mapConfigs.stream().map(MapConfig::getEvictionConfig)
+                .filter(evictionConfig -> evictionConfig.getEvictionPolicy() != EvictionPolicy.NONE).count();
     }
 
     private long countMapWithNativeInMemoryFormat() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getInMemoryFormat() == InMemoryFormat.NATIVE).count();
+        return mapConfigs.stream().map(MapConfig::getInMemoryFormat)
+                .filter(inMemoryFormat -> inMemoryFormat == InMemoryFormat.NATIVE).count();
+    }
+
+    static Predicate<MapConfig> isMapStoreEnabled() {
+        return p -> p.getMapStoreConfig().isEnabled();
+    }
+
+    static Predicate<MapConfig> isMapStoreDisabled() {
+        return p -> !p.getMapStoreConfig().isEnabled();
+    }
+
+    static class LatencyInfo {
+        private long totalLatency;
+        private long totalOperation;
+
+        LatencyInfo(long totalLatency, long totalOperation) {
+            this.totalLatency = totalLatency;
+            this.totalOperation = totalOperation;
+        }
+
+        LatencyInfo setLatency(long totalLatency, long totalOperation) {
+            this.totalLatency = totalLatency;
+            this.totalOperation = totalOperation;
+            return this;
+        }
+
+        LatencyInfo sum(LatencyInfo latencyInfo) {
+            return setLatency(this.totalLatency + latencyInfo.totalLatency,
+                    this.totalOperation + latencyInfo.totalOperation);
+        }
+
+        Long calculateAverage(LatencyInfo latencyInfo) {
+            return latencyInfo.totalOperation == 0 ? -1 : (latencyInfo.totalLatency / latencyInfo.totalOperation);
+        }
+    }
+
+
+    private long mapPutLatency(Node node, Predicate<MapConfig> predicate) {
+
+        LatencyInfo latencyInfo = mapConfigs.stream()
+                .filter(predicate)
+                .map(mapConfig -> node.hazelcastInstance.getMap(mapConfig.getName()).getLocalMapStats())
+                .filter(mapStats -> mapStats.getPutOperationCount() != 0L)
+                .map(mapStats -> info.setLatency(mapStats.getTotalPutLatency(), mapStats.getPutOperationCount()))
```

Comment:
As far as I understand you use the `info` field to temporarily hold two values (total latency and operation count). The problem is that it introduces shared mutable state, which is just error-prone (and it works only because the process is single-threaded - imagine how would this work if `mapPutLatency()` and `mapGetLatency()` would be executed concurrently). 

So instead of the `map()` and `reduce()` operations I suggest
 - initializing `latencyInfo` as a local variable before the stream operations (both fields set to 0)
 - adding a `forEach(stats -> latencyInfo.add(stats.getTotalPutLatency(), stats.getPutOperationCount())`
 
 (assuming that there is an `LatencyInfo#add()` method which works similarly as the current `LatencyInfo#sum()` method)